### PR TITLE
Notify MatrixChat of teamToken after login

### PR DESCRIPTION
**Pull request by lukebarnard1**
Possibly fixes https://github.com/vector-im/riot-web/issues/3299

### CODE REVIEW

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -693,6 +696,16 @@ module.exports = React.createClass({
     },
 
     /**
+     * Called when the team token is acquired at registration or login. This
+     * should only happen if the configured RTS (see config.json teamServerURL)
+     * recognises the user as having an email address that matches a team.
+     */
+    _onSetTeamToken: function(teamToken) {
+        this._teamToken = teamToken;
+        this._setPage(PageTypes.HomePage);
```
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: Is it necessary to set the page here? It's not intuitive to me that it should jump to your welcome page when a  team token is set

Comment (lukebarnard1) [inline src/components/structures/MatrixChat.js]: Maybe this function needs a rename then? `onTeamTokenLogin` ?

**Comment by lukebarnard1** [comment]
This is less racy now - passing `teamToken` around in the same control flow as the main logging-in flow.

File: src/Lifecycle.js [hunk 2]
```diff
@@ -318,7 +317,12 @@ export function setLoggedIn(credentials) {
 
     MatrixClientPeg.replaceUsingCreds(credentials);
 
-    dis.dispatch({action: 'on_logged_in'});
+    teamPromise.then((teamToken) => {
```
Comment (dbkr) [inline src/Lifecycle.js]: done

**Pull request merged**